wire/msgtx.go

```diff
@@ -378,6 +378,13 @@ func (msg *MsgTx) TxHash() chainhash.Hash {
 	return chainhash.DoubleHashH(msg.cachedSeralizedNoWitness)
 }
 
+// WipeCache removes the cached serialized bytes of the transaction. This is
+// useful to be able to get the correct txid after mutating a transaction's
+// state.
+func (msg *MsgTx) WipeCache() {
+	msg.cachedSeralizedNoWitness = nil
```

Comment:
Don't we also need to call this in the helper methods like `AddTxIn()` and `AddTxOut()`? Or is the assumption that `TxHash()` would in practice only be called once the transaction is fully built, so the cache doesn't need to be invalidated?

I'm mostly worrying about uses of `MsgTx` outside of `btcd`, where I'm not sure we can 100% guarantee that we're always using this pattern...